.github/workflows/container-image-buildah.yml

```diff
@@ -0,0 +1,93 @@
+name: Build porta oci container image
+
+on:
+  workflow_call:
+    inputs:
+      platforms:
+        description: comma-separated list of platforms to build for, e.g. linux/amd64,linux/s390x,linux/ppc64le
+        default: linux/amd64
+        type: string
+      custom_tag:
+        description: optional custom tag on remote repo you want image to be tagged with
+        required: false
+        default: ''
+        type: string
+    secrets:
+      QUAY_USER_NAME:
+        required: false
+      QUAY_PASSWORD:
+        required: false
+
+env:
+  # Use docker.io for Docker Hub if empty
+  REGISTRY: quay.io
+  # github.repository as <account>/<repo>
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  buildah:
+    runs-on: ubuntu-latest
+    steps:
+      # Allow multi-target builds
+      # - name: Set up QEMU
+      #  uses: docker/setup-qemu-action@v2
+      #  with:
+      #    platforms: ${{ inputs.platforms }}
+
+      - name: Log in to Red Hat Registry
+        uses: redhat-actions/podman-login@v1
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ secrets.QUAY_USER_NAME }}
+          password: ${{ secrets.QUAY_PASSWORD }}
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          tags: |
+            type=schedule
+            type=ref,event=branch,enable=${{ github.ref_name != 'master' && inputs.custom_tag == '' }}
+            type=raw,value=latest,enable=${{ github.ref_name == 'master' }}
+            type=raw,value=nightly,enable=${{ github.ref_name == 'master' }}
+            ${{ inputs.custom_tag }}
+            type=ref,event=tag
+            type=ref,event=pr
```

Comment:
Not sure it can be brief :)
This is the doc https://github.com/docker/metadata-action#tags-input

But in short, it should apply the branch label if this is not `master` branch and no custom tag provided.

And apply `nightly` and `latest` if it is `master tag.

Also apply whatever `custom_tag` contains if it was set.

The rest are the defaults. 

Also can't be completely tested without merging but some of the things explained above worked.

Reply:
Correction, if you trigger manually, then you have to specify a tag or it will default to `scratch`. I think this will not be the case by other triggers. But have to merge to try out.